smartsim/_core/generation/generator.py

```diff
@@ -24,219 +24,183 @@
-from ..control import Manifest
-from .modelwriter import ApplicationWriter
 
 logger = get_logger(__name__)
 logger.propagate = False
 
 
 class Generator:
-    """The primary job of the generator is to create the file structure
-    for a SmartSim experiment. The Generator is responsible for reading
-    and writing into configuration files as well.
+    """The primary job of the Generator is to create the directory and file structure
+    for a SmartSim Job. The Generator is also responsible for writing and configuring
+    files into the Job directory.
     """
 
-    def __init__(
-        self, gen_path: str, overwrite: bool = False, verbose: bool = True
-    ) -> None:
-        """Initialize a generator object
-
-        if overwrite is true, replace any existing
-        configured applications within an ensemble if there
-        is a name collision. Also replace any and all directories
-        for the experiment with fresh copies. Otherwise, if overwrite
-        is false, raises EntityExistsError when there is a name
-        collision between entities.
-
-        :param gen_path: Path in which files need to be generated
-        :param overwrite: toggle entity replacement
-        :param verbose: Whether generation information should be logged to std out
+    def __init__(self, root: os.PathLike[str]) -> None:
+        """Initialize a Generator object
+
+        The class handles symlinking, copying, and configuration of files
+        associated with a Jobs entity. Additionally, it writes entity parameters
+        used for the specific run into the "smartsim_params.txt" settings file within
+        the Jobs log folder.
         """
-        self._writer = ApplicationWriter()
-        self.gen_path = gen_path
-        self.overwrite = overwrite
-        self.log_level = DEBUG if not verbose else INFO
+        self.root = root
+        """The root path under which to generate files"""
 
-    @property
-    def log_file(self) -> str:
+    def log_file(self, log_path: os.PathLike[str]) -> os.PathLike[str]:
         """Returns the location of the file
-        summarizing the parameters used for the last generation
-        of all generated entities.
+        summarizing the parameters used for the generation
+        of the entity.
 
-        :returns: path to file with parameter settings
+        :param log_path: Path to log directory
+        :returns: Path to file with parameter settings
         """
-        return join(self.gen_path, "smartsim_params.txt")
+        return pathlib.Path(log_path) / "smartsim_params.txt"
 
-    def generate_experiment(self, *args: t.Any) -> None:
-        """Run ensemble and experiment file structure generation
-
-        Generate the file structure for a SmartSim experiment. This
-        includes the writing and configuring of input files for a
-        application.
+    def generate_job(
+        self, job: Job, job_path: os.PathLike[str], log_path: os.PathLike[str]
+    ) -> None:
```

Comment:
method does not use instance attrs, could be a `classmethod`

Reply:
If this object is composed entirely of `staticmethod` and `classmethod`s, does the generator even need to be a class? It's not holding any state and could just be a namespace.